test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -370,5 +377,22 @@ spec = Test.group "Dataflow Warnings" <|
         a.remove_warnings . should_equal 42
         b.remove_warnings . should_equal 42
 
+    Test.specify "should not automatically propagate from suspended arguments" <|
+        x1 = 33
+        x2 = Warning.attach "WARN" 44
+        x3 = Error.throw (Illegal_State.Error "ERR")
+
+        condition_1 x y = if x then y else 42
+        condition_2 x y = if x then y + 100 else 42
+
+        (extract_warnings_or_errors <| condition_1 False x1) . should_equal []
```

Comment:
I think we should be able to use the existing testing infrastructure, that helper essentially does what `asume_no_problems` already did.

```suggestion
        Problems.assume_no_problems <| condition_1 False x1
```

Reply:
Thanks, didn't know about it